client/src/main/kotlin/no/nav/su/se/bakover/client/oppgave/OppgaveHttpClient.kt

```diff
@@ -179,23 +190,63 @@ internal class OppgaveHttpClient(
                     oppgave.right()
                 } else {
                     log.error("Feil ved hent av oppgave $oppgaveId. status=${it.statusCode()} body=${it.body()}")
-                    KunneIkkeSøkeEtterOppgave.left()
+                    OppgaveFeil.KunneIkkeSøkeEtterOppgave.left()
                 }
             }
         }.mapLeft { throwable ->
             log.error("Feil i kallet mot oppgave.", throwable)
-            KunneIkkeSøkeEtterOppgave
+            OppgaveFeil.KunneIkkeSøkeEtterOppgave
         }.flatten()
     }
 
+    private fun oppdaterOppgave(
+        oppgaveId: OppgaveId,
+        token: String,
+        beskrivelse: String,
+    ): Either<OppgaveFeil.KunneIkkeOppdatereOppgave, Unit> {
+        return hentOppgave(oppgaveId, token).mapLeft {
+            OppgaveFeil.KunneIkkeOppdatereOppgave
+        }.flatMap {
+            if (it.erFerdigstilt()) {
+                log.info("Oppgave $oppgaveId kunne ikke oppdateres fordi den allerede er ferdigstilt")
+                OppgaveFeil.KunneIkkeOppdatereOppgave.left()
+            } else {
+                oppdaterOppgave(it, token, beskrivelse).map { }.mapLeft {
```

Comment:
Trenger vi `.map { }` her?

Reply:
Tror det er en kortversjon for .map{Unit}, kan være Any ikke blir casta til Unit